sql/driver/driver_test.go

```diff
@@ -202,6 +204,50 @@ func TestPlaceholders(t *testing.T) {
 	}
 }
 
+func TestConnectionSettings(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	s := server.StartTestServer(nil)
+	url := "https://root@" + s.ServingAddr() + "?certs=test_certs"
+	// Create a new Database called t.
+	db, err := sql.Open("cockroach", url)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := db.Exec(`CREATE DATABASE t`); err != nil {
+		t.Fatal(err)
+	}
+	_ = db.Close()
+
+	// Open a new db client against the server with
+	// a fresh set of connections with the new settings.
+	// We create a database 't' above because you cannot
+	// use sql: "SET DATABASE = t" without the database
+	// existing.
+	url += "&database=t"
+	if db, err = sql.Open("cockroach", url); err != nil {
+		t.Fatal(err)
+	}
+	defer cleanup(s, db)
+	var wq sync.WaitGroup
+	numRoutines := 10
+	wq.Add(numRoutines)
```

Comment:
I'm guessing this is attempting to use a couple of database connections. A more robust way of doing that is to use `db.Tx()` which grab a connection for the lifetime of the transaction.